RecoMuon/L3TrackFinder/src/HLTMuonL2SelectorForL3IO.cc

```diff
@@ -1,78 +1,108 @@
-		l2Src_(consumes<reco::TrackCollection>(iConfig.getParameter<edm::InputTag>("l2Src"))),
-		l3OISrc_(consumes<reco::TrackCollection>(iConfig.getParameter<edm::InputTag>("l3OISrc"))),
-		useOuterHitPosition_(iConfig.getParameter<bool>("useOuterHitPosition")),
-		xDiffMax_(iConfig.getParameter<double>("xDiffMax")),
-		yDiffMax_(iConfig.getParameter<double>("yDiffMax")),
-		zDiffMax_(iConfig.getParameter<double>("zDiffMax")),
-		dRDiffMax_(iConfig.getParameter<double>("dRDiffMax")){
-	LogTrace("Muon|RecoMuon|HLTMuonL2SelectorForL3IO")<<"constructor called";
-	produces<reco::TrackCollection>();
+HLTMuonL2SelectorForL3IO::HLTMuonL2SelectorForL3IO(const edm::ParameterSet& iConfig):  
+  l2Src_(consumes<reco::TrackCollection>(iConfig.getParameter<edm::InputTag>("l2Src"))),
+  l3OISrc_(consumes<reco::RecoChargedCandidateCollection>(iConfig.getParameter<edm::InputTag>("l3OISrc"))),
+  l3linkToken_(consumes<reco::MuonTrackLinksCollection>(iConfig.getParameter<edm::InputTag>("InputLinks"))),
+  applyL3Filters_(iConfig.getParameter<bool>("applyL3Filters")),
+  max_NormalizedChi2_ (iConfig.getParameter<double> ("MaxNormalizedChi2")),
+  max_PtDifference_ (iConfig.getParameter<double> ("MaxPtDifference")),
+  min_Nhits_ (iConfig.getParameter<int> ("MinNhits")),
+  min_NmuonHits_ (iConfig.getParameter<int> ("MinNmuonHits"))  
+{
+   LogTrace("Muon|RecoMuon|HLTMuonL2SelectorForL3IO")<<"constructor called";
+   produces<reco::TrackCollection>();
 }
-  
+
 /// destructor
 HLTMuonL2SelectorForL3IO::~HLTMuonL2SelectorForL3IO(){}
 
 /// create collection of L2 muons not already reconstructed as L3 muons
 void HLTMuonL2SelectorForL3IO::produce(edm::Event& iEvent, const edm::EventSetup& iSetup){
-	const std::string metname = "Muon|RecoMuon|HLTMuonL2SelectorForL3IO";
-  
-//  IN:
-	edm::Handle<reco::TrackCollection> l2TrackCol;
-	iEvent.getByToken(l2Src_, l2TrackCol);
+  const std::string metname = "Muon|RecoMuon|HLTMuonL2SelectorForL3IO";
+	
+  //  IN:
+  edm::Handle<reco::TrackCollection> l2muonH;
+  iEvent.getByToken(l2Src_, l2muonH);
 
-	edm::Handle<reco::TrackCollection> l3TrackCol;
-	iEvent.getByToken(l3OISrc_, l3TrackCol);
+  edm::Handle<reco::RecoChargedCandidateCollection> l3muonH;
+  iEvent.getByToken(l3OISrc_,l3muonH);
 
-//	OUT:
-	auto result = std::make_unique<std::vector<reco::Track>>();
+  // Read Links collection:
+  edm::Handle<reco::MuonTrackLinksCollection> links;
+  iEvent.getByToken(l3linkToken_, links);  
+  
+  //	OUT:
+  std::unique_ptr<reco::TrackCollection > result(new reco::TrackCollection());
+    
+  for (unsigned int il2=0; il2 != l2muonH->size(); ++il2){
+    reco::TrackRef l2muRef(l2muonH, il2);
+    bool re_do_this_L2=true;
+    
+    for (unsigned int il3=0; il3 != l3muonH->size(); ++il3){
+      reco::RecoChargedCandidateRef cand(l3muonH,il3);
+      reco::TrackRef tk = cand->track();
+      
+      bool useThisLink = false;
+      for (unsigned int l(0); l <links->size(); ++l){
```

Comment:
I think it's better to use " l <links->size() && !useThisLink" and remove the subsequent line
"if (useThisLink) continue;"